libs/prelude/Prelude/Functor.idr

```diff
@@ -22,3 +23,5 @@ infixl 4 <$>
 (<$>) : Functor f => (func : a -> b) -> f a -> f b
 func <$> x = map func x
 
+Functor (Pair a) where
+  map f (x, y) = (x, f y)
```

Comment:
You assume that a pair is always of type (a, a) and that you can apply the function to both elements of the pair. However it is frequently (c, a) with map f (c, a) = (c, f a). That last element is picked in a tuple is AFAIK a convention, could as well be the first.